lib/github_deployment.rb

```diff
@@ -0,0 +1,136 @@
+# frozen_string_literal: true
+
+# this class is responsible for handling the GitHub deployments through
+# the API
+class GithubDeployment
+  require 'octokit'
+
+  def initialize(access_token, repository, ref)
+    @client = Octokit::Client.new(access_token: access_token)
+    @repository = repository
+    @ref = ref
+  end
+
+  def lock
+    deployment = latest_deployment
+
+    if deployment.nil?
+      create_a_deployment_and_lock
+      return
+    end
```

Comment:
I think you can do it shorter like this:
```suggestion
    return create_a_deployment_and_lock if deployment.nil? 
```